pkgs/development/python-modules/pijuice/default.nix

```
@@ -0,0 +1,41 @@
+{ lib, fetchFromGitHub, buildPythonPackage, pythonOlder,
+smbus-cffi, urwid }:
+
+buildPythonPackage rec {
+  pname = "pijuice";
+  version = "1.7";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "PiSupply";
+    repo = "PiJuice";
+    # Unfortunately this project doesn't tag releases. This commit altered the
+    # __version__ field in pijuice.py
+    rev = "3ba6719ab614a3dc7495d5d9c900dd4ea977c7e3";
+    sha256 = "GoNN07YgVaktpeY5iYDbfpy5fxkU1x0V3Sb1hgGAQt4=";
+  };
+
+  patches = [
+    ./patch-shebang.diff
+  ];
+
+  PIJUICE_BUILD_BASE = 1;
+
+  sourceRoot = "source/Software/Source";
+
+  propagatedBuildInputs = [ smbus-cffi urwid ];
+
+  preFixup = ''
+    rm $out/bin/pijuice_cli
+    rm $out/bin/pijuice_sys.py
+    rm $out/bin/pijuiceboot
+    mv $out/bin/pijuice_cli.py $out/bin/pijuice_cli
+   '';
+
+  meta = with lib; {
+    description = "PiJuice library";
```

A more substantial description:

```suggestion
    description = "Library and resources for PiJuice HAT for Raspberry Pi";
```